api/tests/test_initialize_user.rs

```diff
@@ -0,0 +1,197 @@
+#![deny(clippy::all)]
```

Comment:
Could you read through these tests cases and see if you think we're missing any? I think I got all the edges cases